src/common/structures/chord.py

```diff
@@ -68,5 +78,46 @@ def get_pitches(self) -> FrozenSet[Pitch]:
             ]
         )
 
-    def get_V7(self):
+    def get_V7(self) -> "Chord":
         return Chord(self.root + Interval.from_str("5"), ChordQuality.Dom7)
+    
+    def to_index(self) -> int:
+        """
+        Returns the Viterbi index of the chord.
+
+        :return: The Viterbi index.
+        """
+        return self.root.value % 12 * len(ChordQuality) + list(ChordQuality).index(self.quality)
+    
+    @classmethod
+    def from_index(cls, index: int) -> "Chord":
+        """
+        Returns a chord from its Viterbi index.
+
+        :param index: The Viterbi index.
+        :return: The chord.
+        """
+        root = index // 3
+        quality = list(ChordQuality)[index % 3]
+        return Chord(Pitch(root), quality)
+
+    @classmethod
+    def from_str(cls, s: str) -> "Chord":
+        """
+        Constructs a chord from a string representation.
+
+        Examples: "C", "Gm", "F#7"
+
+        :param s: The string representation.
+        :return: The chord.
+        """
+        split = 1
+        if len(s) > 1 and s[1] in ['b', '#']:
+            split = 2
+
+        return Chord(Pitch.from_str(s[:split]), ChordQuality.from_str(s[split:]))
+    
+    def __str__(self) -> str:
```

Comment:
Let's leave a note to improve upon this in the future (e.g. return Bb if in the Bb scale).